pylint/checkers/utils.py

```diff
@@ -1653,7 +1652,7 @@ def get_subscript_const_value(node: nodes.Subscript) -> nodes.Const:
     return inferred
 
 
-def get_import_name(importnode: ImportNode, modname: str | None) -> str | None:
+def get_import_name(importnode: nodes.ImportNode, modname: str | None) -> str | None:
```

Comment:
This shouldn't work. `ImportNode` is not part of the `nodes` module.

Reply:
Hmm; it's true I don't see it in astroid.nodes's API. But the tests are green.

Reply:
That is because `mypy` doesn't actually check `astroid` since it has no `py.typed` file.

Reply:
Ha, I thought the import would have failed but it does not because it's using a namespace and not a direct import (``nodes.ImportNode``, with ``from astroid import nodes`` not ``from astroid.nodes import ImportNode``.